src/frontend/src/hooks/UseTable.tsx

```diff
@@ -65,37 +71,71 @@ export function useTable(tableName: string, idAccessor = 'pk'): TableState {
   // Total record count
   const [recordCount, setRecordCount] = useState<number>(0);
 
-  const [pageSizeLoaded, setPageSizeLoaded] = useState<boolean>(false);
-
-  // Pagination data
   const [page, setPage] = useState<number>(1);
-  const [pageSize, setPageSize] = useLocalStorage<number>({
-    key: 'inventree-table-page-size',
-    defaultValue: 25,
-    sync: false,
-    deserialize: (value: string | undefined) => {
-      setPageSizeLoaded(true);
-      return value === undefined ? 25 : JSON.parse(value);
-    }
-  });
 
-  const [hiddenColumnsLoaded, setHiddenColumnsLoaded] =
-    useState<boolean>(false);
-
-  // A list of hidden columns, saved to local storage
-  const [hiddenColumns, setHiddenColumns] = useLocalStorage<string[] | null>({
-    key: `inventree-hidden-table-columns-${tableName}`,
-    defaultValue: null,
-    sync: false,
-    deserialize: (value) => {
-      setHiddenColumnsLoaded(true);
-      return value === undefined ? null : JSON.parse(value);
-    }
+  const [storedDataLoaded, setStoredDataLoaded] = useState<boolean>(false);
+
+  const [tableData, setTableData] = useState<StoredTableData>({
+    pageSize: 25,
+    hiddenColumns: null
   });
 
-  const storedDataLoaded = useMemo(() => {
-    return pageSizeLoaded && hiddenColumnsLoaded;
-  }, [pageSizeLoaded, hiddenColumnsLoaded]);
+  const [storedTableData, setStoredTableData] =
+    useLocalStorage<StoredTableData>({
+      key: `inventree-table-data-${tableName}`,
+      getInitialValueInEffect: true,
+      // sync: false,  // Do not use this option - see below
+      defaultValue: {
+        pageSize: 25,
+        hiddenColumns: null
+      },
+      deserialize: (value: any) => {
+        const tableData =
+          value === undefined
+            ? {
+                pageSize: 25,
+                hiddenColumns: null
+              }
+            : JSON.parse(value);
+
+        if (!storedDataLoaded) {
+          setStoredDataLoaded((wasLoaded: boolean) => {
+            if (!wasLoaded) {
+              // First load of stored table data - copy to local state
+              // We only do this on first load, to avoid live syncing between tabs
+              // Note: The 'sync: false' option is not used, it does not perform as expected
+              setTableData(tableData);
+            }
+            return true;
+          });
+        }
+        return tableData;
+      }
+    });
```

Comment:
this looks pretty similar to a zustand store; could we use that (as it is proven to work)

Reply:
Yeah we should look into that as an alternative option to the `useLocalStorage` hook from mantine. This is working OK now but might be cleaner to migrate to zustand.